tests/cpp/test_memory.cpp

```diff
@@ -352,7 +361,7 @@ void scheduleTile(
   }
 }
 
-TEST_P(TMALdstTest, SimpleLoad) {
+TEST_P(TMASimpleLdstTest, Load) {
```

Comment:
Example kernel:

```CUDA
__global__ void nvfuser_none_f0_c0_r0_g0(Tensor<__half, 1, 1> T0, const __grid_constant__ TensorMap var0, Tensor<__half, 1, 1> T2) {
  alignas(16) extern __shared__ char array[];
  const unsigned smem_offset = 0;
  const TensorMap* ptr1;
  ptr1 = &var0;
  nvfuser_index_t i2;
  i2 = 8LL * ((nvfuser_index_t)blockIdx.x);
  Array<nvfuser_index_t, 1, 1> a3;
  a3 = Array<nvfuser_index_t, 1, 1>{i2};
  nvfuser_index_t i4;
  i4 = ((nvfuser_index_t)threadIdx.x) + i2;
  __half* T1 = reinterpret_cast<__half*>(array + smem_offset + 0);
  uint64_t* T3 = reinterpret_cast<uint64_t*>(array + smem_offset + 16LL);
  mbarrier::init(toSmem(T3), 1U);
  __syncthreads();
  if ((((nvfuser_index_t)threadIdx.x) == 0)) {
    uint64_t i5;
    i5 = mbarrier::arriveExpectTX(toSmem(T3), 16U);
    Hopper::cpAsyncBulkTensorTileG2S((Hopper::CpAsyncBulkTensorTileG2SIndex<1>{ ptr1, a3, toSmem(T3) }), toSmem(T1));
    mbarrier::wait(toSmem(T3), i5);
  }
  __syncthreads();
  mbarrier::inval(toSmem(T3));
  __syncthreads();
  if ((i4 < T0.logical_size[0LL])) {
    T2[i4]
       = T1[((nvfuser_index_t)threadIdx.x)];
  }
}
```

Reply:
Is the listing correct?

In the test we check if predicated is for a single thread, but in listing mbarrier code is predicated for 4 threads:
```
if ((((nvfuser_index_t)threadIdx.x) < 4)) {
```

Reply:
Good catch! It is actually

```CUDA
__global__ void nvfuser_none_f0_c0_r0_g0(Tensor<__half, 1, 1> T0, const __grid_constant__ TensorMap var0, Tensor<__half, 1, 1> T2) {
  alignas(16) extern __shared__ char array[];
  const unsigned smem_offset = 0;
  const TensorMap* ptr1;
  ptr1 = &var0;
  nvfuser_index_t i2;
  i2 = 8LL * ((nvfuser_index_t)blockIdx.x);
  Array<nvfuser_index_t, 1, 1> a3;
  a3 = Array<nvfuser_index_t, 1, 1>{i2};
  nvfuser_index_t i4;
  i4 = ((nvfuser_index_t)threadIdx.x) + i2;
  __half* T1 = reinterpret_cast<__half*>(array + smem_offset + 0);
  uint64_t* T3 = reinterpret_cast<uint64_t*>(array + smem_offset + 16LL);
  mbarrier::init(toSmem(T3), 1U);
  __syncthreads();
  if ((((nvfuser_index_t)threadIdx.x) == 0)) {
    uint64_t i5;
    i5 = mbarrier::arriveExpectTX(toSmem(T3), 16U);
    Hopper::cpAsyncBulkTensorTileG2S((Hopper::CpAsyncBulkTensorTileG2SIndex<1>{ ptr1, a3, toSmem(T3) }), toSmem(T1));
    mbarrier::wait(toSmem(T3), i5);
  }
  __syncthreads();
  mbarrier::inval(toSmem(T3));
  __syncthreads();
  if ((i4 < T0.logical_size[0LL])) {
    T2[i4]
       = T1[((nvfuser_index_t)threadIdx.x)];
  }
}
```
I copy-pasted the wrong kernel...

Reply:
Thanks for checking this, I though I misunderstood something :)